src/tools/perf/lib/libperf.c

```diff
@@ -1267,6 +1282,164 @@ static ucs_status_t ucp_perf_test_send_local_data(ucx_perf_context_t *perf,
+        ucs_error("failed to send AM on ep %p: %s\n", dmn_ep,
+                  ucs_status_string(UCS_PTR_STATUS(sptr)));
+        ucs_free(dmn_info);
+        return UCS_PTR_STATUS(sptr);
+    }
+
+    return UCS_OK;
+}
+
+static ucs_status_t ucp_perf_setup_daemon_endpoints(ucx_perf_context_t *perf)
+{
+    /* Sender and receiver roles for DPU daemons are defined by perftest
+     * group_index property */
+    unsigned group_index      = rte_call(perf, group_index);
+    unsigned peer_group_index = rte_peer_index(rte_call(perf, group_size),
+                                               group_index);
+    int is_local_sender       = (group_index % 2) != 0;
+    int is_remote_sender      = (peer_group_index % 2) != 0;
+    ucp_ep_params_t ep_params = {};
+    struct sockaddr_storage *local_dmn_addr;
+    struct sockaddr_storage *remote_dmn_addr;
+    ucs_status_t status;
+
+    if (!perf->params.ucp.is_daemon_mode) {
+        return UCS_OK;
+    }
+
+    ucs_assert_always(is_local_sender != is_remote_sender);
+    ucs_assert_always(perf->params.thread_count == 1);
+
+    if (is_local_sender) {
+        local_dmn_addr  = &perf->params.ucp.dmn_local_addr;
+        /* IP and port of remote daemon peer for subsequent PEER_INIT, which
+         * is triggered on the sender side
+         */
+        remote_dmn_addr = &perf->params.ucp.dmn_remote_addr;
+    } else {
+        local_dmn_addr  = &perf->params.ucp.dmn_remote_addr;
+        /* No need to trigger PEER_INIT on the receiver daemon side */
+        remote_dmn_addr = NULL;
+    }
+
+    /* Create EP towards the local daemon */
+    ep_params.field_mask       = UCP_EP_PARAM_FIELD_FLAGS |
+                                 UCP_EP_PARAM_FIELD_SOCK_ADDR |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLER |
+                                 UCP_EP_PARAM_FIELD_ERR_HANDLING_MODE;
+    ep_params.err_mode         = UCP_ERR_HANDLING_MODE_PEER;
+    ep_params.err_handler.cb   = err_cb;
+    ep_params.err_handler.arg  = NULL;
+    ep_params.flags            = UCP_EP_PARAMS_FLAGS_CLIENT_SERVER;
+    ep_params.sockaddr.addr    = (struct sockaddr*)local_dmn_addr;
+    ep_params.sockaddr.addrlen = sizeof(*local_dmn_addr);
+
+    /* Coverity detects that ep_params.conn_request may be dereferenced in
+     * ucp_ep_create -> ucp_ep_create_api_conn_request, which is not possible
+     * with passed flags and field_mask */
+    /* coverity[var_deref_model] */
+    status = ucp_ep_create(perf->ucp.tctx[0].perf.ucp.worker, &ep_params,
+                           &perf->ucp.tctx[0].perf.ucp.ep);
+    if (status != UCS_OK) {
+        ucs_error("failed to create endpoint: %s", ucs_status_string(status));
+        return status;
+    }
+
+    /* Create and send HOST_INIT message to the local DPU daemon */
+    status = ucp_perf_daemon_init_send(perf, perf->ucp.tctx[0].perf.ucp.ep,
+                                       remote_dmn_addr);
+    if (status != UCS_OK) {
+        /* Error is logged */
```

Comment:
what does this comment mean? I'd remove it

Reply:
It explains why we don't log an error here - because it was logged already in the function
But ok to remove